dialogy/plugins/text/intent_entity_mutator/__init__.py

```diff
@@ -0,0 +1,309 @@
+"""
+    IntentEntityMutatorPlugin
+"""
+
+from typing import Any, List, Optional, Dict, Union, Tuple, cast
+
+from dialogy.base import Input, Output, Plugin, Guard
+from dialogy.types import BaseEntity, Intent
+from dialogy.plugins.text.intent_entity_mutator.registry import transcript_functions_map
+
+from . import const
+
+
+class IntentEntityMutatorPlugin(Plugin):
+    def __init__(
+        self,
+        rules: Dict[str, List[Dict[str, Any]]],
+        guards: Optional[List[Guard]] = None,
+        dynamic_output_path: bool = True,
+        **kwargs: Any,
+    ) -> None:
+
+        if self.validate_rules(rules):
+            self.rules = rules
+        else:
+            raise ValueError("The rules file is not in the correct format")
+
+        super().__init__(
+            guards=guards, dynamic_output_path=dynamic_output_path, **kwargs
+        )
+
+    def validate_rules(self, rules: Dict[str, List[Dict[str, Any]]]) -> bool:
+        rules_base_keys = list(rules.keys())
+
+        if rules_base_keys and "swap_rules" not in rules_base_keys:
+            raise ValueError(
+                f"Did not receive {const.BASE_KEY} as the base key, received {rules_base_keys[0]}"
+            )
+        elif not rules_base_keys:
+            raise ValueError("Did not receive any base_key")
+
+        rules_ = rules[const.BASE_KEY]
+
+        permissible_transcript_functional_condition_keys = list(
+            transcript_functions_map.keys()
+        )
+
+        for rule in rules_:
+            rule_keys = sorted(list(rule.keys()))
+            if rule_keys != const.MANDATORY_RULE_KEYS:
+                raise ValueError(
+                    f"The rule is either missing some keyword or incorrect keyword entered. \
+                        Received: {rule_keys}. Expected: {const.MANDATORY_RULE_KEYS} \
+                            {rule}"
```

Comment:
Wherever we are appending the offensive rule in the exception message, lets start the rule from a new line. I think it would improve the readability of the message.